runtime/src/identity.rs

```diff
@@ -244,15 +247,7 @@ decl_module! {
 
             <DidRecords>::mutate( id,
             |record| {
-                let mut new_signing_ids = signing_ids.iter()
-                    .filter( |s_id| record.signing_identities.iter().find( |id| id == s_id).is_none())
-                    .chain( record.signing_identities.iter())
-                    .cloned()
-                    .collect::<Vec<_>>();
-                new_signing_ids.sort();
-                new_signing_ids.dedup();
-
-                (*record).signing_identities = new_signing_ids;
+                (*record).add_signing_identities( &signing_ids);
```

Comment:
In this implementation, SigningKeys with type Identity do not have to do the "two step process". Will check w/ product, but if we need to implement this, I think you'd need a separate function:
`pub fn authorize_identity_join_to_identity(origin, target_id: IdentityId) -> Result`
which joins `CurrentDid` to the `target_id` (assuming it had been previously authorised)

Reply:
Fixed, signing identities and keys share the "two step process" (and the code related).